src/Sylius/Bundle/ApiBundle/Canceller/PaymentRequestCanceller.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Canceller;
+
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+
+/** @experimental */
+final class PaymentRequestCanceller implements PaymentRequestCancellerInterface
+{
+    /**
+     * @param PaymentRequestRepositoryInterface<PaymentRequestInterface> $paymentRequestRepository
+     */
+    public function __construct(private PaymentRequestRepositoryInterface $paymentRequestRepository)
+    {
+    }
+
+    public function cancelPaymentRequests(string $paymentId, string $paymentMethodCode): void
```

Comment:
How about,

```suggestion
    public function cancelPaymentRequests(string $paymentId, ?string $paymentMethodCode = null): void
```
for case, when order is cancelled from admin panel via "cancel" button, then I want all all payment requests to become cancelled as well and don't partically care to specify which `paymentMethodCode` I want to cancel.

Reply:
This is only API case, as PaymentRequest exists only in API

Reply:
Well, yes, but payment requests should be cancelled from the admin panel on order cancel too, we shouldn't be in situation where order is cancelled, but payment requests still actual, but I guess this is for another PR about PaymentRequest SM usage.

Reply:
Oh ok, sorry I didn't understand correctly 😅, well yeah probably you are right with the SM